src/libcrun/utils.c

```diff
@@ -447,7 +447,9 @@ crun_safe_ensure_at (bool dir, int dirfd, const char *dirpath, size_t dirpath_le
           TEMP_FAILURE_RETRY (close (wd_cleanup));
           wd_cleanup = -1;
         }
-      wd_cleanup = cwd;
+      else {
+        wd_cleanup = cwd;
```

Comment:
why the new else block?  `wd_cleanup` must be assigned in any case

Reply:
It looks like you already have a replacement of this setup in master, so I've adopted that.